zilliqa/src/crypto.rs

```diff
@@ -20,20 +22,44 @@ use sha3::{Digest, Keccak256};
 
 /// The signature type used internally in consensus, to e.g. sign block proposals.
 #[derive(Debug, Clone, Copy, PartialEq)]
-pub struct NodeSignature(bls_signatures::Signature);
+pub struct NodeSignature(Signature<Bls12381G2Impl>);
 
 impl NodeSignature {
     pub fn identity() -> NodeSignature {
-        NodeSignature(G2Affine::identity().into())
+        // Default to Basic signatures - it's the normal signature.
+        NodeSignature(Signature::<Bls12381G2Impl>::Basic(
+            blsful::inner_types::G2Projective::identity(),
+        ))
     }
 
     pub fn from_bytes(bytes: &[u8]) -> Result<NodeSignature> {
-        Ok(NodeSignature(bls_signatures::Signature::from_bytes(bytes)?))
+        // Default to Basic signatures - it's the normal signature.
+        Ok(NodeSignature(Signature::<Bls12381G2Impl>::Basic(
+            blsful::inner_types::G2Projective::from_compressed(bytes.try_into()?)
+                .expect("blst_p2_uncompress() error"),
```

Comment:
Should we propagate the error instead? `from_bytes` is returning a `Result`, but this implementation never returns an `Err`. It seems like `from_compressed` will fail if `!p.is_on_curve()` or `!p.is_torsion_free()` and potentially for more reasons if the underlying C library doesn't return success.